lib/ModuleReason.js

```diff
@@ -4,9 +4,45 @@
 */
 "use strict";
 
-module.exports = class ModuleReason {
+const util = require("util");
+
+class ModuleReason {
 	constructor(module, dependency) {
 		this.module = module;
 		this.dependency = dependency;
+		this._chunks = null;
+	}
+
+	hasChunk(chunk) {
+		if(this._chunks) {
+			if(this._chunks.has(chunk))
+				return true;
+		} else if(this.module._chunks.has(chunk))
+			return true;
+		return false;
+	}
+
+	rewriteChunks(oldChunk, newChunks) {
+		if(!this._chunks) {
+			if(!this.module._chunks.has(oldChunk))
+				return;
+			this._chunks = new Set(this.module._chunks);
+		}
+		if(this._chunks.has(oldChunk)) {
+			this._chunks.delete(oldChunk);
+			newChunks.forEach(newChunk => this._chunks.add(newChunk));
+		}
+	}
+}
+
+Object.defineProperty(ModuleReason.prototype, "chunks", {
```

Comment:
Unsure if this is completely necessary, or if it should or should not be deprecated, but copied the strategy from the other set migrations.